### ISSUE TITLE
关于update_pi重参数化的问题

### ISSUE THREAD

**Issue by Arya87** [molumitu/BOOM_MBRL#2]
您好，在boom_alg.py的第261~264行中，代码实际做的操作是：用当前策略采样出的新动作 pis，减去 Buffer 里老策略（Planner）的均值 mu，再除以当前策略的标准差 std，算出一个混合的epsilon。
但是按照论文中Eq.2的公式，不应该是Buffer里的动作，减去策略Pi的均值，再除以策略Pi的std吗？还是我理解有错？您能帮我解答一下吗，谢谢！

**Comment by lilinfuxle1** [comment]
我做过一些尝试如果是动作aβ会变成bc，对planner的模仿强度太强反而导致会收敛到前期的局部最优，论文的Likelihood-free alignment metric感觉描述得有点像bc但是实际上是软约束，在q认为好的地方（即mppi表现好的地方）轻微对齐mppi的均值

**Comment by Arya87** [comment]
> 我做过一些尝试如果是动作aβ会变成bc，对planner的模仿强度太强反而导致会收敛到前期的局部最优，论文的Likelihood-free alignment metric感觉描述得有点像bc但是实际上是软约束，在q认为好的地方（即mppi表现好的地方）轻微对齐mppi的均值

感谢解答~
软约束的解释的话那我可以理解了。
另外我还发现一个问题，在boom_alg.py的250~266行
_, pis, log_pis, log_std = self.model.pi(zs, task)
qs = self.model.Q(zs, pis, task, return_type="min")
self.scale.update(qs[0])
qs = self.scale(qs)
					
forward_kl = self.scale(forward_kl) if self.scale.value > 2.0 else torch.zeros_like(forward_kl)
forward_kl = torch.softmax(qs.detach().squeeze(),dim=-1) * forward_kl

这里的Q weight应该是计算的aβ的Q值，但是实际代码里却是计算的actor action的Q值

**Comment by lilinfuxle1** [comment]
> > 我做过一些尝试如果是动作aβ会变成bc，对planner的模仿强度太强反而导致会收敛到前期的局部最优，论文的Likelihood-free alignment metric感觉描述得有点像bc但是实际上是软约束，在q认为好的地方（即mppi表现好的地方）轻微对齐mppi的均值
> 
> 感谢解答~ 软约束的解释的话那我可以理解了。 另外我还发现一个问题，在boom_alg.py的250~266行 _, pis, log_pis, log_std = self.model.pi(zs, task) qs = self.model.Q(zs, pis, task, return_type="min") self.scale.update(qs[0]) qs = self.scale(qs)
> 
> forward_kl = self.scale(forward_kl) if self.scale.value > 2.0 else torch.zeros_like(forward_kl) forward_kl = torch.softmax(qs.detach().squeeze(),dim=-1) * forward_kl
> 
> 这里的Q weight应该是计算的aβ的Q值，但是实际代码里却是计算的actor action的Q值

他的论文我感觉描述的不是很好,总感觉是planner决定什么好然后蒸馏给policy,但是实际上是policy决定什么地方好然后用planner的分布去稳固policy的分布，我感觉其实和上面的软约束一样，都是防止太早收敛为planner的样子，保留一些探索性
